usr/sbin/decompress-kernel

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+
+KERNEL_PATH="$1"
+
+LZ4_PATTERN=$(printf '\002!L\030')
+
+if grep -aq "${LZ4_PATTERN}" "${KERNEL_PATH}"; then
+	echo "Kernel is compressed with LZ4, decompressing..." >&2
+	tmp_file=$(mktemp /tmp/raw-kernel-XXXX)
+	trap "rm -rf ${tmp_file}" 0 2
```

Comment:
```suggestion
	trap "rm -f ${tmp_file}" 0 2
```
Should be enough.

And I would add a check for `mktemp` to ensure we do not delete something unexpected (even if I doubt anyone will have this happen ever).

Reply:
Yes it will be enough indeed. I added a check for the result of the mktemp call, and a dependency to coreutils / mktemp packages (even though those are essential packages and should be installed already).